features/steps/basic_replication.py

```diff
@@ -90,3 +97,8 @@ def replication_works(context, primary, replica, time_limit):
         When I add the table test_{0} to {1}
         Then table test_{0} is present on {2} after {3} seconds
     """.format(int(time()), primary, replica, time_limit))
+
+@then('there are {num_nodes:d} nodes after {time_limit:d} seconds')
+def check_node_count(context, num_nodes, time_limit):
+    assert context.pctl.check_node_count(num_nodes, time_limit),\
+        "There are not {0} nodes in the system after {1} seconds".format(num_nodes, time_limit)
```

Comment:
IMO, what we really want to check that the node refuse to start with a very specific error message.
It will simplify a lot of things and make unnecessary many changes introduced in the `features/environment.py`.

Reply:
Yes this is definitely better, removed what was bloating `features/environment.py` and replaced with a check that looks for a specific error from the duplicate node